tests/integration/api/test_event_view_operations.py

```diff
@@ -729,6 +741,122 @@ def assert_datetime_almost_equal(s1: pd.Series, s2: pd.Series):
     assert (s1 - s2).dt.total_seconds().abs().max() <= 1e-6
 
 
+@pytest.mark.parametrize("input_format", ["table", "uploaded_table"])
+@pytest.mark.asyncio
+async def test_get_historical_features__feature_table_cache(
+    session,
+    data_source,
+    feature_group,
+    feature_group_per_category,
+    user_entity,
+    new_user_id_entity,
+    input_format,
+    feature_table_cache_metadata_service,
+):
+    """Test feature table cache create/update"""
+    _ = user_entity, new_user_id_entity
+
+    # feature table cache is only supported for observation tables
+    assert input_format in {"table", "uploaded_table"}
+
+    feature_list_1 = FeatureList(
+        [
+            feature_group["COUNT_2h"],
+            feature_group_per_category["COUNT_BY_ACTION_24h"],
+            feature_group_per_category["MOST_FREQUENT_ACTION_24h"],
+        ],
+        name="My FeatureList 1",
+    )
+
+    feature_list_2 = FeatureList(
+        [
+            feature_group["COUNT_2h"],
+            feature_group["COUNT_24h"],
+            feature_group_per_category["COUNT_BY_ACTION_24h"],
+            feature_group_per_category["ENTROPY_BY_ACTION_24h"],
+            feature_group_per_category["MOST_FREQUENT_ACTION_24h"],
+            feature_group_per_category["NUM_UNIQUE_ACTION_24h"],
+            feature_group_per_category["ACTION_SIMILARITY_2h_to_24h"],
+        ],
+        name="My FeatureList 2",
+    )
+
+    df_training_events, df_historical_expected = get_training_events_and_expected_result()
+    if "table" in input_format:
+        df_historical_expected.insert(0, "__FB_TABLE_ROW_INDEX", np.arange(1, 11))
+
+    observation_table = await create_observation_table_from_dataframe(
+        session,
+        df_training_events,
+        data_source,
+    )
+
+    historical_feature_table_name = f"historical_feature_table_{ObjectId()}"
+    historical_feature_table = feature_list_1.compute_historical_feature_table(
+        observation_table,
+        historical_feature_table_name,
+    )
+    df_historical_features_1 = await get_dataframe_from_materialized_table(
+        session, historical_feature_table
+    )
```

Comment:
Can we also call `historical_feature_table.to_pandas()` to make sure downloading the table works? Note it's expected that the result in `to_pandas()` doesn't include the row index column.